container/openejb-core/src/main/java/org/apache/openejb/core/mdb/MdbContainer.java

```diff
@@ -505,4 +574,139 @@ public static BeanContext current() {
         private TransactionPolicy txPolicy;
         private ThreadContext oldCallContext;
     }
+
+    private static class MdbActivationContext {
+        private final ClassLoader classLoader;
+        private final BeanContext beanContext;
+        private final ResourceAdapter resourceAdapter;
+        private final EndpointFactory endpointFactory;
+        private final ActivationSpec activationSpec;
+
+        private AtomicBoolean started = new AtomicBoolean(false);
+
+        public MdbActivationContext(final ClassLoader classLoader, final BeanContext beanContext, final ResourceAdapter resourceAdapter, final EndpointFactory endpointFactory, final ActivationSpec activationSpec) {
+            this.classLoader = classLoader;
+            this.beanContext = beanContext;
+            this.resourceAdapter = resourceAdapter;
+            this.endpointFactory = endpointFactory;
+            this.activationSpec = activationSpec;
+        }
+
+        public ResourceAdapter getResourceAdapter() {
+            return resourceAdapter;
+        }
+
+        public EndpointFactory getEndpointFactory() {
+            return endpointFactory;
+        }
+
+        public ActivationSpec getActivationSpec() {
+            return activationSpec;
+        }
+
+        public boolean isStarted() {
+            return started.get();
+        }
+
+        public void start() throws ResourceException {
+            if (started.get()) {
+                return;
+            }
+
+            final ClassLoader oldCl = Thread.currentThread().getContextClassLoader();
+            Thread.currentThread().setContextClassLoader(classLoader);
```

Comment:
try/finally for such operation is important to not mess up a reused thread (spent 3 days to fix such an issue in our arquillian adapter so know it looks trivial to fix but hard to find when it pops up)